### PR TITLE
Coroutine promise types _must_ have member unhandled_exception in C++20

### PR DESCRIPTION
**Pull request by CaseyCarter**
... even when `_CPPUNWIND` is not defined.

### CODE REVIEW

File: stl/inc/experimental/generator [hunk 1]
```diff
@@ -55,19 +56,29 @@ namespace experimental {
...
+#if defined(_CPPUNWIND) || defined(__cpp_impl_coroutine)
             void unhandled_exception() noexcept {
+#if 1 // TRANSITION, VSO-1172852
+#ifdef _CPPUNWIND
                 _Eptr = _STD current_exception();
-            }
+#else // ^^^ _CPPUNWIND / !_CPPUNWIND vvv
+                abort();
 #endif // _CPPUNWIND
+#else // ^^^ workaround / no workaround
```
Comment (mnatsuhara) [inline stl/inc/experimental/generator:67]: Are you missing `vvv` here? 😸 (Ditto on line 133.)

Comment (CaseyCarter) [reply stl/inc/experimental/generator:67]:
```suggestion
#else // ^^^ workaround / no workaround vvv
```

File: stl/inc/experimental/generator [hunk 2]
```diff
@@ -117,7 +128,11 @@ namespace experimental {
             iterator& operator++() {
                 _Coro.resume();
                 if (_Coro.done()) {
+#if 1 // TRANSITION, VSO-1172852
                     _STD exchange(_Coro, {}).promise()._Rethrow_if_exception();
+#else // ^^^ workaround / no workaround
```
Comment (CaseyCarter) [inline stl/inc/experimental/generator:133]:
```suggestion
#else // ^^^ workaround / no workaround vvv
```

**Comment by StephanTLavavej** [comment]
Thanks for fixing this bug - it may be legacy/experimental tech, but a bug is a bug. :beetle: :smiley_cat:

**Pull request merged**